modules/oe_newsroom_newsletter/tests/src/Functional/SubscribeNewsletterTest.php

```diff
@@ -0,0 +1,186 @@
+    $page->pressButton('Subscribe');
+    $assertSession->pageTextContains('The subscription service is not configured at the moment. Please try again later.');
+
+    // Tests after setting the newsroom configuration.
+    $this->configureNewsroom();
+
+    // Test missing e-mail address and unchecked privacy statement (subscribe).
+    $this->drupalGet($this->subscribePath);
+    $assertSession->pageTextContains('This is the introduction text.');
+    $assertSession->linkByHrefExists('/privacy-url');
+    $assertSession->linkExists('privacy statement');
+    $page->hasField('Your e-mail');
+    $page->hasUncheckedField('By checking this box, I confirm that I want to register for this service, and I agree with the privacy statement');
+    $page->pressButton('Subscribe');
+    $assertSession->pageTextContains('Your e-mail field is required.');
+    $assertSession->pageTextContains('By checking this box, I confirm that I want to register for this service, and I agree with the privacy statement field is required.');
+
+    // Test invalid e-mail address (subscribe).
+    $this->drupalGet($this->subscribePath);
+    $page->fillField('Your e-mail', '@example.com');
+    $page->checkField('By checking this box, I confirm that I want to register for this service, and I agree with the privacy statement');
+    $page->pressButton('Subscribe');
+    $assertSession->pageTextContains('The email address @example.com is not valid.');
+
+    // Subscribe the newsletter.
+    $this->drupalGet($this->subscribePath);
+    $page->fillField('Your e-mail', 'mail@example.com');
+    $page->checkField('By checking this box, I confirm that I want to register for this service, and I agree with the privacy statement');
+    $page->pressButton('Subscribe');
+    $assertSession->pageTextContains('Thanks for Signing Up to the service: Test Newsletter Service');
+
+    // Test missing e-mail address (unsubscribe).
+    $this->drupalGet($this->unsubscribePath);
+    $page->hasField('Your e-mail');
+    $page->pressButton('Unsubscribe');
+    $assertSession->pageTextContains('Your e-mail field is required.');
+
+    // Test invalid e-mail address (unsubscribe).
+    $this->drupalGet($this->unsubscribePath);
+    $page->fillField('Your e-mail', '@example.com');
+    $page->pressButton('Unsubscribe');
+    $assertSession->pageTextContains('The email address @example.com is not valid.');
+
+    // Unsubscribe the newsletter.
+    $this->drupalGet($this->unsubscribePath);
+    $page->fillField('Your e-mail', 'mail@example.com');
+    $assertSession->hiddenFieldValueEquals('distribution_list', '123');
+    $page->pressButton('Unsubscribe');
+    $assertSession->pageTextContains('Successfully unsubscribed!');
+  }
+
+  /**
+   * Test the subscription form if a user can subscribe to a newsletter twice.
+   *
+   * @group oe_newsroom_newsletter
+   */
+  public function testSubscribeNewsletterTwice(): void {
```

Comment:
We could merge this method with the multiple one, and just place two blocks.

Reply:
not sure if I am correctly getting your idea because I am not seeing the advantages for that :) could you clarify your thought, please?